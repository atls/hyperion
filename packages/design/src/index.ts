export * from './Sample'
export * from './Layout'
export * from './Container'
export * from './List'
export * from './Text'
export * from './Space'
