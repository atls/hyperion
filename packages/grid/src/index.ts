export * from './Column'
export * from './Grid'
