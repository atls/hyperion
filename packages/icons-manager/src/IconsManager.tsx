import React, { FC, useEffect, useState } from 'react'

import styled                             from '@emotion/styled'

interface Props {
  color?: string
  hoverColor?: string
  clickedColor?: string
  children?: any
  theme: any
}

const IconUI = styled.div({
  cursor: 'pointer',
  display: 'inline-flex',
})

export const IconsManager: FC<Props> = ({ theme, children, hoverColor, clickedColor, color }) => {
  const themeColor = theme.colors[color] || color
  const themeHoverColor = theme.colors[hoverColor] || hoverColor
  const themeClickedColor = theme.colors[clickedColor] || clickedColor

  const [currentColor, setCurrentColor] = useState(themeColor)

  useEffect(() => {
    setCurrentColor(themeColor)
  }, [themeColor])

  return (
    <IconUI
      onMouseEnter={() => setCurrentColor(themeHoverColor || themeColor)}
      onMouseLeave={() => setCurrentColor(themeColor)}
      onMouseDown={() => setCurrentColor(themeClickedColor || themeColor)}
      onMouseUp={() => setCurrentColor(themeColor)}
    >
      {React.cloneElement(children, { key: 'a', color: currentColor })}
    </IconUI>
  )
}
