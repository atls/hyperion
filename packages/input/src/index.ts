export * from './Textarea'
export * from './Input'
export * from './Checkbox'
