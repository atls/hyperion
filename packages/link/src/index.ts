export * from './Link'
export * from './NextLink'
export * from './RouteLink'
