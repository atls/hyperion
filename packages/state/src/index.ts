export * from './withState'
export * from './useActive'
export * from './useHover'
export * from './useFocus'
