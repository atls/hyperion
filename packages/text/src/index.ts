export * from './Text'
export * from './Space'
export * from './SearchResult'
