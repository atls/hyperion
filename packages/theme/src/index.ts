import * as theme from './theme'

export * from './Provider'
export * from './globals'

export { theme }
