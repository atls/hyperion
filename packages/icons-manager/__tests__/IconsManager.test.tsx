import React            from 'react'
import renderer         from 'react-test-renderer'

import { IconsManager } from '../src/index'

it('IconsManager should render correctly', () => {
  const defaultTheme = {
    colors: {
      green: '#0f0',
      black: '#000',
      white: '#fff',
      blue: '#00f',
      red: '#f00',
      lightGray: 'gray',
    },
    shadows: {
      controls: '0 0 10px rgba(0,0,0,0.5)',
    },
    borders: {
      controls: '2px solid black',
    },
  }

  const Icon = ({ color = 'red' }) => <h1 style={{ color: `${color}` }}>I can change color</h1>

  const tree = renderer
    .create(
      <IconsManager color='red' hoverColor='yellow' clickedColor='black'>
        <Icon />
      </IconsManager>,
    )
    .toJSON()

  expect(tree).toMatchSnapshot()
})
