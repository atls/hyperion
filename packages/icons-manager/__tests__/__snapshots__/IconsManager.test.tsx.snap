// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconsManager should render correctly 1`] = `
.emotion-0 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

<div
  className="emotion-0"
  onMouseDown={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
>
  <h1
    style={
      Object {
        "color": "#f00",
      }
    }
  >
    I can change color
  </h1>
</div>
`;
